aea/protocols/dialogue/base.py

```diff
@@ -1309,6 +1344,121 @@ def teardown(self) -> None:
             return
         self._dump()
 
+    def remove(self, dialogue_label: DialogueLabel) -> None:
+        """Remove dialogue from memory and persistent storage."""
+        if dialogue_label in self._terminal_state_dialogues_labels:
+            collection = self._terminal_dialogues_collection
+        else:
+            collection = self._active_dialogues_collection
+
+        super().remove(dialogue_label)
+
+        if collection:
+            collection.remove(str(dialogue_label))
+
+
+class PersistDialoguesStorageWithOffloading(PersistDialoguesStorage):
+    """Dialogue Storage with dialogues offlosding."""
+
+    def dialogue_terminal_state_callback(self, dialogue: "Dialogue") -> None:
+        """Call on dialogue reaches terminal staste."""
+        if not self._terminal_dialogues_collection:  # pragma: nocover
+            # no storage available, fallback
+            super().dialogue_terminal_state_callback(dialogue)
+            return
+        # do offloading
+        # push to storage
+        self._terminal_dialogues_collection.put(
+            str(dialogue.dialogue_label), dialogue.json()
+        )
+        # remove from memory
+        self.remove(dialogue.dialogue_label)
+
+    def get(self, dialogue_label: DialogueLabel) -> Optional[Dialogue]:
+        """Try to get dialogue by label from memory or persists storage."""
+        dialogue = super().get(dialogue_label)
+        if dialogue:
+            return dialogue
+
+        dialogue = self._get_dialogue_from_collection(
+            dialogue_label, self._terminal_dialogues_collection
+        )
+        if dialogue:
+            # get dialogue from terminal state collection and cache it
+            self._add_terminal_state_dialogue(dialogue)
```

Comment:
why?

Reply:
just to avoid direct access to internal valirables